lib/orderbook/TradingPair.ts

```diff
@@ -267,6 +267,16 @@ class TradingPair {
     return maps.buy.get(orderId) || maps.sell.get(orderId);
   }
 
+  public addOrderHold = (orderOrOrderId: string | StampedOwnOrder, holdAmount: number) => {
```

Comment:
it's indeed don't make much sense to have this method accept an order instance, and perform the same logic as previously `OrderBook` did. I think it should only accept `orderId`, even though you have the order instance sometimes. Otherwise it's still breaking the encapsulation, and reducing `addOrderHold` to a util function. 
The point is not to encourage updates via instance mutations, although they work at the moment. You might get an instance copy for example, and so it won't work, or that other actions need to be applied in the repository (like in `order.quantity = 0`).